src/roswire/ros1/launch/reader.py

```diff
@@ -51,7 +51,7 @@
 def _read_contents(tag: ET.Element) -> str:
     """Reads the text contents of an XML element."""
     # FIXME add support for CDATA -- possibly via lxml or xml.dom?
-    return "".join(t.text for t in tag if t.text)
+    return tag.text if tag.text else ""  # return "".join(t.text for t in tag if t.text)
```

Comment:
I worry a little bit about this breaking existing functionality, but not enough to block the changes before the paper deadline. Have you tried to see if we get the same results as before on existing systems? In any case, it makes sense to remove the commented-out code.

Reply:
I haven't checked...But it is definitely the case that `t.text for t in tag if t.text` doesn't work for the common use of rosparam.